### DISCUSSION TITLE
open /dev/null: operation not permitted when importing videos (convert to jpeg)

### DISCUSSION THREAD

**Discussion by remisharrock** [photoprism/photoprism#2898] [Q&A]
I run photoprism in docker as root, photos are importing OK but mp4 videos have this error :
import: open /dev/null: operation not permitted in 2022/11/20221113_141607_DC5FF8F8.mp4 (convert to jpeg)

I see the mp4 files in the originals subdirectories after import but don't see them in photoprism 

THanks for the help.

**Comment by lastzero** [comment]
Might be an issue with ffmpeg if it's only happening with videos. Exact cause can not be determined without any details on your specific environment.

**Reply by remisharrock** [reply]
All right so as I said I use docker, especially docker compose with photoprism/photoprism:latest  for the photoprism image composed with mariadb. I'm currently seeing version 220901-f493607b0 of photoprism.
Docker is running inside an lxc container Debian 10 as root user. My volumes are correctly mounted first on the lxc then on the photoprism docker container. Do you have an idea how to test if ffmpeg is doing something wrong ?inside the photoprism container I guess ?

**Reply by lastzero** [reply]
No custom config options? Hardware transcoding? Before you try anything else, I recommend you update to our latest version as described in our docs:

https://docs.photoprism.app/getting-started/updates/#docker-compose

Since nobody else has such problems, it's either related to a custom config or your LXC environment (which we cannot see or test).

**Reply by remisharrock** [reply]
thanks for the reply; I updated to Build 221116-122ebfb70   
Here is my docker compose file https://pastebin.com/QXct0MyN
I deleted the folder created with my latest import in the originals (2022/11), I then re-indexed this folder to update the database (in the logs I saw that the files were correctly removed)
I then re-imported the files and this time I got this error for all mp4 files:

Error
November 17, 2022 at 9:56:06 AM GMT+1
import: file type mp4 not supported in 2022/11/20221111_161845_788530CE.mp4 (convert to jpeg)

in my lxc (which is running docker), I store the originals here:
pwd
/zb/photoprism/originals/

and this folder is mounted like this:
mount | grep zb
zb on /zb type zfs (rw,xattr,noacl)

what I noticed is that with the old version 220901-f493607b0 files where written with those permissions:
-rwxrwxr-x 1 root root   4344056 Nov 15 16:42 20221103_135203_5ACF8DF6.jpg

and now with this new version 221116-122ebfb70  with those permissions:
-rw-rw-r-- 1 root root   4344056 Nov 17 09:55 20221103_135203_5ACF8DF6.jpg

I don't use custom options I think ?
![image](https://user-images.githubusercontent.com/8048877/202402948-643a1a8c-0fc8-4c41-bb99-b2481f9aaffa.png)
![image](https://user-images.githubusercontent.com/8048877/202403668-c002c56d-f324-404c-b4c8-8ee33e799155.png)
![image](https://user-images.githubusercontent.com/8048877/202403759-870a6788-c20b-4489-8b29-4fd7b9cf0879.png)
![image](https://user-images.githubusercontent.com/8048877/202403790-8734a360-5790-4b2b-9106-37ec808ae121.png)

any more idea ? Information you need to help me ?
Thank you again

**Reply by remisharrock** [reply]
maybe another info:
I went "inside" the docker (docker exec -it photoprism-photoprism-1 /bin/bash) and "ffprobe" an mp4 file inside the import directory and saw those two interesting sentences at the end:

Unsupported codec with id 0 for input stream 0
Unsupported codec with id 0 for input stream 3 

By the way the files were created with a Google pixel 7 phone.

here is the full output:

root@221116-jammy:/photoprism/import/pixel7$ ffprobe PXL_20221104_143703839.TS.mp4
ffprobe version 4.4.2-0ubuntu0.22.04.1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 11 (Ubuntu 11.2.0-19ubuntu1)
  configuration: --prefix=/usr --extra-version=0ubuntu0.22.04.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  WARNING: library configuration mismatch
  avcodec     configuration: --prefix=/usr --extra-version=0ubuntu0.22.04.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared --enable-version3 --disable-doc --disable-programs --enable-libaribb24 --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libtesseract --enable-libvo_amrwbenc --enable-libsmbclient
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'PXL_20221104_143703839.TS.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 131072
    compatible_brands: isomiso2mp41
    creation_time   : 2022-11-04T14:37:20.000000Z
    location        : +43.6462+1.3540/
    location-eng    : +43.6462+1.3540/
    com.android.capture.fps: 30.000000
  Duration: 00:00:13.97, start: 0.000000, bitrate: 19810 kb/s
  Stream #0:0(eng): Data: none (mett / 0x7474656D), 44 kb/s (default)
    Metadata:
      creation_time   : 2022-11-04T14:37:20.000000Z
      handler_name    : MetaHandle
  Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 191 kb/s (default)
    Metadata:
      creation_time   : 2022-11-04T14:37:20.000000Z
      handler_name    : SoundHandle
      vendor_id       : [0][0][0][0]
  Stream #0:2(eng): Video: hevc (Main) (hvc1 / 0x31637668), yuvj420p(pc, bt709), 1920x1080, 19566 kb/s, SAR 1:1 DAR 16:9, 30 fps, 30 tbr, 90k tbn, 90k tbc (default)
    Metadata:
      creation_time   : 2022-11-04T14:37:20.000000Z
      handler_name    : VideoHandle
      vendor_id       : [0][0][0][0]
  Stream #0:3(eng): Data: none (mett / 0x7474656D) (default)
    Metadata:
      creation_time   : 2022-11-04T14:37:20.000000Z
      handler_name    : MetaHandle
Unsupported codec with id 0 for input stream 0
Unsupported codec with id 0 for input stream 3

**Reply by lastzero** [reply]
If you believe that it is a problem with the file, we need a sample to find the issue and fix it if necessary: https://photoprism.app/contact

Note that we can only provide individual support to sponsors as our time is very limited.

**Reply by remisharrock** [reply]
All right, here is a sample
https://transfert.free.fr/56RXlc
I might also try to debug myself, try to fix it and do a pull request one day ;)

**Reply by lastzero** [reply]
Works perfectly:

```
DEBU[2022-11-17T10:23:58Z] video: Debug/mp4/PXL_20221109_180140359.TS.mp4 is HVC1 compressed and cannot be streamed directly, average bitrate 21.9 MBit/s 
DEBU[2022-11-17T10:23:58Z] metadata: PXL_20221109_180140359.TS.mp4 uses utc by default (video/mp4) 
INFO[2022-11-17T10:23:58Z] libx264: transcoding Debug/mp4/PXL_20221109_180140359.TS.mp4 to avc 
INFO[2022-11-17T10:24:15Z] libx264: created PXL_20221109_180140359.TS.mp4.avc [16.470601691s] 
DEBU[2022-11-17T10:24:15Z] server: GET /api/v1/videos/498683378bfa9fc0366b7d374251007b79e56566/28i6llwe/avc (206) [16.5261175s] 
```

**Reply by remisharrock** [reply]
Ok I see you transcoded the video, In my config I see "Disable FFMPEG" don't transcode videos with FFMPEG. I'll try unchecking this.

**Reply by remisharrock** [reply]
it works ! 
Thank you ;)
